test/image/mocks/legend_visibility.json

```diff
@@ -1,24 +1,90 @@
 {
-    "data":[
-        {"name":"bar A","type":"bar","y":["2","3","5","4","1"]},
-        {"name":"bar B","type":"bar","visible":"legendonly","y":["3","5","7","6","8"]},
-        {"name":"bar C","type":"bar","y":["2","4","6","3","8"]},
-        {"name":"line A","yaxis":"y2","y":["2","3","5","4","1"]},
-        {"name":"line B","yaxis":"y2","visible":"legendonly","y":["3","5","7","6","8"]},
-        {"name":"line C","yaxis":"y2","y":["2","4","6","3","8"]},
-        {"name":"box A","type":"box","xaxis":"x2","yaxis":"y3","y":[1,2,3,4,5,6,7,8,9]},
-        {"name":"box B","type":"box","xaxis":"x2","yaxis":"y3","visible":"legendonly","y":[2,3,3,4,4,4,5,5,6]},
-        {"name":"box C","type":"box","xaxis":"x2","yaxis":"y3","y":[1,4,6,7,7,8,8,8,9]}
-    ],
-    "layout": {
-        "showlegend":true,
-        "height":400,
-        "width":600,
-        "barnorm":"fraction",
-        "barmode":"stack",
-        "xaxis":{"domain":[0,0.45]},
-        "xaxis2":{"anchor":"y3","domain":[0.55,1]},
-        "yaxis2":{"overlaying":"y","side":"right"},
-        "yaxis3":{"anchor":"x2"}
+  "data": [
+    {
+      "name": "bar A",
+      "type": "bar",
+      "y": ["2", "3", "5", "4", "1"]
+    },
+    {
+      "name": "bar B",
+      "type": "bar",
+      "visible": "legendonly",
+      "y": ["3", "5", "7", "6", "8"]
+    },
+    {
+      "name": "bar C",
+      "type": "bar",
+      "y": ["2", "4", "6", "3", "8"]
+    },
+    {
+      "name": "line A",
+      "yaxis": "y2",
+      "y": ["2", "3", "5", "4", "1"]
+    },
+    {
+      "name": "line B",
+      "yaxis": "y2",
+      "visible": "legendonly",
+      "y": ["3", "5", "7", "6", "8"]
+    },
+    {
+      "name": "line C",
+      "yaxis": "y2",
+      "y": ["2", "4", "6", "3", "8"]
+    },
+    {
+      "name": "box A",
+      "type": "box",
+      "xaxis": "x2",
+      "yaxis": "y3",
+      "y": [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    },
+    {
+      "name": "box B",
+      "type": "box",
+      "xaxis": "x2",
+      "yaxis": "y3",
+      "visible": "legendonly",
+      "y": [2, 3, 3, 4, 4, 4, 5, 5, 6]
+    },
+    {
+      "name": "box C",
+      "type": "box",
+      "xaxis": "x2",
+      "yaxis": "y3",
+      "y": [1, 4, 6, 7, 7, 8, 8, 8, 9]
+    },
+    {
+      "name": "marker A",
+      "mode": "markers",
+      "visible": "legendonly",
+      "y": [1, 4, 6, 3, 2, 8, 8, 8, 4],
+      "yaxis": "y2",
+      "marker": {
+        "size": 15,
+        "symbol": ["square", "square", "square", "square", "square", "square", "square", "square", "square"]
```

Comment:
The legend item here currently renders as a circle (using the `Drawing.singlePointStyle` fallback.